docs/source/developers/java/building.rst

```diff
@@ -171,11 +171,31 @@ IDE Configuration
 IntelliJ
 --------
 
-To start working on Arrow in IntelliJ, just open the `java/`
-subdirectory of the Arrow repository.
-
+To start working on Arrow in IntelliJ: build the project once from the command
+line using ``mvn clean install``. Then open the ``java/`` subdirectory of the
+Arrow repository, and update the following settings:
+
+* In the Files tool window, find the path ``vector/target/generated-sources``,
+  right click the directory, and select Mark Directory as > Generated Sources
+  Root. There is no need to mark other generated sources directories, as only
+  the ``vector`` module generates sources.
```

Comment:
If this is needed we need to enforce "to keep a directory marked as a source root after it has been deleted/recreated through": Preferences > Build, Execution, Deployment > Build Tools > Maven > Importing > `Keep source and test folders on project reload`

FYI, I am not able to test this step "Mark Directory as > Generated Sources Root" on my local machine, it finished without errors.

Reply:
You won't get errors until you open a file that references one of the generated classes or do an _IntelliJ_ build (not a Maven build)

Reply:
Oh, ok, thank you.